tests/functional/relation_tests/test_relation_type_change.py

```diff
@@ -34,31 +40,86 @@ def error_message(self) -> str:
+    Model(models.ICEBERG_INCREMENTAL_TABLE, "table", "iceberg", incremental=True),
 ]
 scenarios = [Scenario(*scenario) for scenario in product(relations, relations)]
 
 
 class TestRelationTypeChange:
 
+    @staticmethod
+    def include(scenario) -> bool:
+        return (
+            scenario.initial.table_format != "iceberg" and scenario.final.table_format != "iceberg"
+        )
+
     @pytest.fixture(scope="class", autouse=True)
     def seeds(self):
         return {"my_seed.csv": models.SEED}
 
     @pytest.fixture(scope="class", autouse=True)
     def models(self):
-        yield {f"{scenario.name}.sql": scenario.initial.model for scenario in scenarios}
+        yield {
+            f"{scenario.name}.sql": scenario.initial.model
+            for scenario in scenarios
+            if self.include(scenario)
+        }
 
     @pytest.fixture(scope="class", autouse=True)
     def setup(self, project):
         run_dbt(["seed"])
         run_dbt(["run"])
         for scenario in scenarios:
-            update_model(project, scenario.name, scenario.final.model)
+            if self.include(scenario):
+                update_model(project, scenario.name, scenario.final.model)
         run_dbt(["run"])
 
     @pytest.mark.parametrize("scenario", scenarios, ids=[scenario.name for scenario in scenarios])
     def test_replace(self, project, scenario):
-        relation_type = query_relation_type(project, scenario.name)
-        assert relation_type == scenario.final.relation_type, scenario.error_message
+        if self.include(scenario):
+            relation_type = query_relation_type(project, scenario.name)
+            assert relation_type == scenario.final.relation_type, scenario.error_message
+            if relation_type == "dynamic_table":
+                dynamic_table = describe_dynamic_table(project, scenario.name)
+                assert dynamic_table.catalog.table_format == scenario.final.table_format
+        else:
+            pytest.skip()
+
+
+"""
+Upon adding the logic needed for seamless transitions to and from incremental models without data loss, we can coalesce these test cases.
+"""
+
+
+class TestRelationTypeChangeIcebergOn(TestRelationTypeChange):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"flags": {"enable_iceberg_materializations": True}}
+
+    @staticmethod
+    def include(scenario) -> bool:
+        return (
+            (
+                scenario.initial.table_format == "iceberg"
+                or scenario.final.table_format == "iceberg"
+            )
+            and not scenario.initial.incremental
+            and not scenario.final.incremental
+        )
+
+
+class TestRelationTypeChangeBetweenIcebergMaterializations(TestRelationTypeChange):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"flags": {"enable_iceberg_materializations": True}}
+
+    @staticmethod
+    def include(scenario) -> bool:
```

Comment:
I would roll this scenario into `TestRelationTypeChangeIcebergOn` by removing the `and not scenario.initial.incremental and not scenario.final.incremental` from `include`. The project configuration is the same, so we're really just expending what gets run in the scenario.

The goal of combining all of these into a single test case was to minimize how many times we need to call `dbt run`, since that was the bottleneck for these tests. The reason for two test cases was to support the two different project configuration (the behavior flag). The base test only runs scenarios where iceberg is not a factor. The "iceberg on" test only runs scenarios where iceberg is a factor. Taken together, it's just the 16 (4 starting states X 4 ending states) possible scenarios.